Source/Controllers/DataControllers/SPTableData.m

```diff
@@ -1028,26 +1130,31 @@ - (BOOL)updateStatusInformationForCurrentTable
 			tableStatusResult = [mySQLConnection queryString:[NSString stringWithFormat:@"SELECT COUNT(1) FROM %@", [escapedTableName backtickQuotedString] ]];
 			// this query can fail e.g. if a table is damaged
 			if (tableStatusResult && ![mySQLConnection queryErrored]) {
-				[status setObject:[[tableStatusResult getRowAsArray] objectAtIndex:0] forKey:@"Rows"];
+				[status safeSetObject:[[tableStatusResult getRowAsArray] safeObjectAtIndex:0] forKey:@"Rows"];
```

Comment:
If you stop using objectAtIndex for index 0, but start using `firstObject`, you will save a lot of time for yourself :)

Reply:
It's not me using ... it's existing code! But yes, I should check if index is 0 when fixing these crashes.

Reply:
so go ahead and change them all?

Reply:
Don't have to do it all, but `objectAtIndex:0` should be replaced for safety... up to you :)